### DISCUSSION TITLE
Is it possible to make reset shortcut for date range picker?

### DISCUSSION THREAD

**Discussion by Ryandi-S** [rsuite/rsuite#2066] [Q&A] [answered]
Hello.
I'm using date picker & date range picker from rsuite 5.0.3. My team want a reset/clean button like this :
<img width="255" alt="image" src="https://user-images.githubusercontent.com/93305443/139188502-4c03df22-036e-4635-a1ee-d7791504e71e.png">

I can make it work with the date picker, bit not with the date range picker. It always got disable like this :
<img width="499" alt="image" src="https://user-images.githubusercontent.com/93305443/139188678-232a6608-acde-4d09-be9f-8b7aec63c73d.png">

Is anyone know how to make reset/clean button using the ranges prop for date range picker? Thanks in advance

**Comment by SevenOutman** [comment, marked as answer]
Either `DatePicker` or `DateRangePicker` has a `cleanable` prop which, when set `true`, displays a "clear" button on the text box that can erase the selected value when clicked on. Does that satisfy what you want from a "reset" button?

**Reply by Ryandi-S** [reply]
Thanks for the response :)
Yes i tried it. The clear button that appear on the textbox works fine. But I requested to make the clear button to be below the picker, that's why i'm trying to put the reset button using `ranges` prop.

The code i used for `DatePicker` is as follow :
```
ranges={[
  {
    label: 'Reset',
    value: null
  }
]}
```
It works fine. I tried similiar with the `DateRangePicker` :
```
ranges={[
  {
    label: 'Reset',
    value: [null,null]
  }
]}
```
but it doesn't work. The button appears, but rendered as disabled (as in my previous `DateRangePicker` screenshot).

Is there something wrong from my code, or is it not possible to erase `DateRangePicker` value this way?

**Reply by simonguo** [reply]
This happens because invalid values are checked in DateRangePicker.
https://github.com/rsuite/rsuite/blob/main/src/DateRangePicker/DateRangePicker.tsx#L617

**Reply by Ryandi-S** [reply]
Ah, I see... so, it's not possible to pass an empty date value without making the button disabled?

**Reply by simonguo** [reply]
This is a requirement that we did not consider in our design. But there is a temporary solution.
```tsx
const App = () => {
  const [value, setValue] = useState([]);
  return (
    <DateRangePicker
      value={value}
      ranges={[
        {
          label: 'clear',
          value: [1, 1]
        }
      ]}
      onChange={value => {
        if (value[0] === 1 && value[1] === 1) {
          setValue([]);
        } else {
          setValue(value);
        }
      }}
    />
  );
};

```

**Reply by Ryandi-S** [reply]
That works well for my needs! Thank you very much :)
